mesonbuild/backend/ninjabackend.py

```diff
@@ -865,11 +865,8 @@ def create_target_linker_introspection(self, target: build.Target, linker: T.Uni
             tgt[lnk_hash] = lnk_block
 
     def generate_target(self, target):
-        try:
-            if isinstance(target, build.BuildTarget):
-                os.makedirs(self.get_target_private_dir_abs(target))
-        except FileExistsError:
-            pass
+        if isinstance(target, (build.BuildTarget, build.CustomTarget)):
+            os.makedirs(self.get_target_private_dir_abs(target), exist_ok=True)
```

Comment:
Yeah, please do not unconditionally do this for all targets, it would be better to only create the directory if `@PRIVATE_DIR@` appears in its `command:` kwarg.

Reply:
Ok, took a different approach where it's only created when replacing `@PRIVATE_DIR@`. This has the added advantage of working across all backends.